File: src/main/java/com/moovie/backend/domain/usecase/movies/GetMovieReviews.java
package com.moovie.backend.domain.usecase.movies;

import com.moovie.backend.domain.repository.MoviesRepository;
import com.moovie.backend.domain.model.MovieReviewListing;
import org.springframework.stereotype.Service;

@Service
public class GetMovieReviews {

    private final MoviesRepository moviesRepository;

    public GetMovieReviews(MoviesRepository moviesRepository) {
        this.moviesRepository = moviesRepository;
    }

    public MovieReviewListing invoke(int movieId, int page) {
        return moviesRepository.getMovieReviews(movieId, page);
    }
}

File: src/main/kotlin/org/example/domain/usecase/movies/GetMovieReviews.kt
package org.example.domain.usecase.movies

import kotlinx.coroutines.runBlocking
import org.example.domain.repository.MoviesRepository
import org.example.model.MovieReviewListing

class GetMovieReviews(private val repository: MoviesRepository) {
    operator fun invoke(page: Int, userId: String? = null, movieId: Int? = null): MovieReviewListing = runBlocking {
        repository.getMovieReviews(page, userId, movieId)
    }
}
